src/typeDefs/User.js

```diff
@@ -10,24 +10,32 @@ export default gql`
   }
 
   extend type Mutation {
-    createUser(userInput: UserInput!): User!
+    login(email: String!, password: String!): User!
+
+    """
+    Do not provide ID while creating user
+    """
+    signUp(userInput: UserInput!): User!
     updateUser(userInput: UserInput!): User!
-    deleteUser(useId: String!): User!
+    deleteUser(userId: String!): User!
   }
 
+  """
+  No need to provide ip during creation, but should have IP while updating
+  """
   input UserInput {
-    id: ID!
-    type: String!
+    id: ID
+    type: UserType!
     firstName: String!
     lastName: String!
     password: String!
     dateOfBirth: String!
     contactNo: String!
     email: String!
-    ip: String!
-    vehicle: VehicleInput!
-    employeeType: String!
-    favouriteBranch: [FavouriteBranchInput!]!
+    ip: String
+    vehicle: [VehicleInput!]!
```

Comment:
mandatory field?
if employee or brand owner also need input vehicle?
you intent to save an empty array is it?

Reply:
You're right. Will change.